### ISSUE TITLE
Corrupt data in struct passed to method via reflection on Unix

### ISSUE THREAD

**Issue by stephentoub** [dotnet/coreclr#7685]
Repro:
```C#
using System;
using System.Reflection;

public class Program
{
    public static void Main(string[] args)
    {
        var r = new RectangleF(1.2f, 3.4f, 5.6f, 7.8f);
        DoStuff(r);
        typeof(Program).GetTypeInfo().GetDeclaredMethod("DoStuff").Invoke(null, new object[] { r });
    }

    public static void DoStuff(RectangleF r)
    {
        Console.WriteLine(r);
    }
}

public struct RectangleF
{
    private float _x, _y, _width, _height;

    public RectangleF(float x, float y, float width, float height)
    {
        _x = x; _y = y; _width = width; _height = height;
    }

    public override string ToString() => $"[{_x}, {_y}, {_width}, {_height}]";
}
```
On CentOS and Ubuntu, this outputs various forms of:
```
[1.2, 3.4, 5.6, 7.8]
[1.2, 3.4, 1.699571E-38, 0]
```
with the reflection-based call getting garbage for two of the floats in the struct.

I got the above repro using the 1.0.1 release, but I started looking into it because of a CI failure in a new test I was adding on top of the latest coreclr pulled into corefx. (The issue that was happening in CI was only happening on CentOS for some reason, not on Ubuntu or OSX.  I've not tried the above repro on OSX.)

**Comment by cmckinsey** [comment]
@stephentoub smells like ABI issue. this is on x64?

**Comment by janvorli** [comment]
Let me dig into that.

**Comment by stephentoub** [comment]
> this is on x64?

Yup

**Comment by janvorli** [comment]
It is really a bug in copying float args to the transition frame in the runtime. It happens when a struct uses two xmm registers. The offset to the next floating point register in the transition frame is shifted by 8 instead of by 16. I have fixed the same issue in the runtime for passing floating point arguments about a year ago, but obviously haven't realized that we have it at this other place.
I will submit a PR with a fix and a related regression test soon.

**Comment by stephentoub** [comment]
Thanks, @janvorli.

**Comment by janvorli** [comment]
Fixed by #7716
